examples/segnet/evaluate.py

```diff
@@ -72,45 +68,27 @@ def main():
     it = chainer.iterators.SerialIterator(test, batch_size=args.batchsize,
                                           repeat=False, shuffle=False)
 
-    n_positive = [0 for _ in range(n_class)]
-    n_true = [0 for _ in range(n_class)]
-    n_true_positive = [0 for _ in range(n_class)]
-    for batch in it:
-        img, gt_label = concat_examples(batch, args.gpu)
-        img = chainer.Variable(img, volatile=True)
-        pred_label = F.argmax(F.softmax(model(img)), axis=1)
-        pred_label = cuda.to_cpu(pred_label.data)
-        gt_label = cuda.to_cpu(gt_label)
-        for cls_i in range(n_class):
-            if cls_i in ignore_labels:
-                continue
-            n_positive[cls_i] += np.sum(pred_label == cls_i)
-            n_true[cls_i] += np.sum(gt_label == cls_i)
-            n_true_positive[cls_i] += np.sum(
-                (pred_label == cls_i) * (gt_label == cls_i))
-
-    ious = []
-    mean_accs = []
-    pixel_accs = []
-    for cls_i in range(n_class):
-        if cls_i in ignore_labels:
-            continue
-        deno = n_positive[cls_i] + n_true[cls_i] - n_true_positive[cls_i]
-        iou = n_true_positive[cls_i] / deno
-        ious.append(iou)
-        print('{:>23} : {:.4f}'.format(camvid_label_names[cls_i], iou))
-
-        mean_accs.append(n_true_positive[cls_i] / n_true[cls_i])
-        pixel_accs.append([n_true_positive[cls_i], n_true[cls_i]])
+    imgs, pred_values, gt_values = apply_prediction_to_iterator(
+        model.predict, it)
+    pred_labels, = pred_values
+    gt_labels, = gt_values
+
+    confusion = calc_semantic_segmentation_confusion(
+        pred_labels, gt_labels, len(camvid_label_names))
```

Comment:
`len(camvid_label_names)` is no longer needed.
https://github.com/chainer/chainercv/blob/master/chainercv/evaluations/eval_semantic_segmentation_iou.py#L7